partiql-spi/src/main/kotlin/org/partiql/spi/function/builtins/internal/Accumulator.kt

```diff
@@ -49,7 +48,7 @@ internal fun checkIsNumberType(funcName: String, value: Datum) {
         return checkIsNumberType(funcName, value.lower())
     }
     if (!value.type.isNumber()) {
-        throw TypeCheckException("Expected NUMBER but received ${value.type}.")
+        throw PErrors.unexpectedTypeException(value.type, listOf(PType.tinyint(), PType.smallint(), PType.integer(), PType.bigint(), PType.decimal(), PType.numeric(), PType.real(), PType.doublePrecision()))
```

Comment:
nit: could have some internal val within SPI that lists out the numeric types to make this less hard-coded

Reply:
not sure if this has been addressed but non-blocking